test/e2e_mc/framework.go

```diff
@@ -0,0 +1,438 @@
+			// kind: ClickHouseInstallation. Retry in such scenario.
+			if strings.Contains(stderr, "ClickHouseInstallation") || strings.Contains(stdout, "ClickHouseInstallation") {
+				return false, nil
+			}
+			return false, fmt.Errorf("error when deploying the flow visibility YML %s: %s, %s, %v", flowVisibilityYML, stdout, stderr, err)
+		}
+		return true, nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// createPodOnNode creates a pod in the test namespace with a container whose type is decided by imageName.
+// Pod will be scheduled on the specified Node (if nodeName is not empty).
+// mutateFunc can be used to customize the Pod if the other parameters don't meet the requirements.
+func (data *MCTestData) createPodOnNode(td *e2e.TestData, name string, ns string, nodeName string, image string, command []string, args []string, env []corev1.EnvVar, ports []corev1.ContainerPort, hostNetwork bool, mutateFunc func(*corev1.Pod)) error {
+	// image could be a fully qualified URI which can't be used as container name and label value,
+	// extract the image name from it.
+	imageName := e2e.GetImageName(image)
+	return data.CreatePodOnNodeInNamespace(td, name, ns, nodeName, imageName, image, command, args, env, ports, hostNetwork, mutateFunc)
+}
+
+// CreatePodOnNodeInNamespace creates a pod in the provided namespace with a container whose type is decided by imageName.
+// Pod will be scheduled on the specified Node (if nodeName is not empty).
+// mutateFunc can be used to customize the Pod if the other parameters don't meet the requirements.
+func (data *MCTestData) CreatePodOnNodeInNamespace(td *e2e.TestData, name, ns string, nodeName, ctrName string, image string, command []string, args []string, env []corev1.EnvVar, ports []corev1.ContainerPort, hostNetwork bool, mutateFunc func(*corev1.Pod)) error {
+	podSpec := corev1.PodSpec{
+		Containers: []corev1.Container{
+			{
+				Name:            ctrName,
+				Image:           image,
+				ImagePullPolicy: corev1.PullIfNotPresent,
+				Command:         command,
+				Args:            args,
+				Env:             env,
+				Ports:           ports,
+			},
+		},
+		RestartPolicy: corev1.RestartPolicyNever,
+		HostNetwork:   hostNetwork,
+	}
+	if nodeName != "" {
+		podSpec.NodeSelector = map[string]string{
+			"kubernetes.io/hostname": nodeName,
+		}
+	}
+	if nodeName == data.controlPlaneNames[td.GetClusterName()] {
+		// tolerate NoSchedule taint if we want Pod to run on control-plane Node
+		podSpec.Tolerations = e2e.ControlPlaneNoScheduleTolerations()
+	}
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"antrea-e2e": name,
+				"app":        ctrName,
+			},
+		},
+		Spec: podSpec,
+	}
+	if mutateFunc != nil {
+		mutateFunc(pod)
+	}
+	if _, err := td.GetClientSet().CoreV1().Pods(ns).Create(context.TODO(), pod, metav1.CreateOptions{}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func flowVisibilityCleanup(tb testing.TB, data *MCTestData) {
```

Comment:
These functions seem quite similar to what we use in the existing test/e2e directory. To collect coverage, there must be a step to kill the daemon processes and aggregate the coverage files. Also, in order to reduce duplicate code, would it be possible to use the same functions in test/e2e (and maybe move them to a test/utils folder) while iterating over each cluster?

Reply:
I've updated these functions and added coverage. Thanks.